### DISCUSSION TITLE
Kitten dependencies

### DISCUSSION THREAD

**Discussion by mushrowan** [kovidgoyal/kitty#7015] [Q&A]
It seems that OOTB Debian doesn't support the SSH kitten, unless I'm doing something very wrong. Is there a list of dependencies for kitten somewhere that I can install to get it working, or are they the same as that of kitty?

**Comment by kovidgoyal** [comment]
There are no dependencies, kitten is a statically compiled executable.

**Reply by mushrowan** [reply]
Weird, I was getting some errors with command not found. When I manually found the kitten executable and tried to run it I got some error related to some lib not being found. Will try to replicate later (I'm at work right now) and post here and maybe it'll bring some silly mistake I'm making to light.

**Reply by mushrowan** [reply]
Really appreciate your work on kitty by the way. If there are any contributions I can make in the python code I'd be happy to contribute sometime.

**Reply by mushrowan** [reply]
it seems that on Debian the ssh kitten doesn't add ~/.local/share/kitty-ssh-kitten/kitty/bin to PATH. Any idea why this may be?
![image](https://github.com/kovidgoyal/kitty/assets/69822445/3fb5b3b2-59d6-4058-a752-7ba931177025)
![image](https://github.com/kovidgoyal/kitty/assets/69822445/c68155e7-1590-4c93-8680-caaa71e17626)

**Reply by kovidgoyal** [reply]
I doubt very much its the kitten not adding the PATH. More likely somethingin your user or system shell rc files is overriding PATH once the shell is launched. The other alternative is if you already have a kitty binary on your system then the kitten will not override it with its own. See the install_kitty_bootstrap() function in bootstrap-utils.sh. You can override this via https://sw.kovidgoyal.net/kitty/kittens/ssh/#opt-kitten-ssh.remote_kitty  in ssh.conf

**Comment by kovidgoyal** [comment]
On Fri, Jan 19, 2024 at 06:32:34AM -0800, ro wrote:
> Really appreciate your work on kitty by the way. If there are any contributions I can make in the python code I'd be happy to contribute sometime.

You're welcome, and generally speaking, if you want to contribute code,
scratch your own itch. If there is some feature/bugfix you'd like to add
to kitty, work on that.
